### PR TITLE
SONAR-6906 Filter rules on profile language when bulk active rules

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-6906 Filter rules on profile language when bulk active rules

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleQueryFactory.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    String sortParam = request.getSort();
+    if (sortParam != null) {
+      query.setSortField(sortParam);
+      query.setAscendingSort(request.getAsc());
+    }
+    return query;
+  }
+
+  @CheckForNull
+  public QualityProfileDto getByKey(String key) {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleQueryFactory.java]: this method shouldn't be in `RuleQueryFactory` or not public.

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleQueryFactory.java]: indeed, it should be private

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java [hunk 2]
```diff
@@ -495,7 +454,7 @@ private static SearchWsRequest toSearchWsRequest(Request request) {
       .setActivation(request.paramAsBoolean(PARAM_ACTIVATION))
       .setActiveSeverities(request.paramAsStrings(PARAM_ACTIVE_SEVERITIES))
       .setAsc(request.mandatoryParamAsBoolean(Param.ASCENDING))
-      .setAvailableSince(request.param(PARAM_AVAILABLE_SINCE))
+      // .setAvailableSince(request.param(PARAM_AVAILABLE_SINCE))
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java]: why comment this line?

File: sonar-ws/src/main/java/org/sonarqube/ws/client/rule/SearchWsRequest.java [hunk 3]
```diff
@@ -78,11 +79,11 @@ public SearchWsRequest setAsc(Boolean asc) {
   }
 
   @CheckForNull
-  public String getAvailableSince() {
+  public Date getAvailableSince() {
     return availableSince;
   }
 
-  public SearchWsRequest setAvailableSince(@Nullable String availableSince) {
+  public SearchWsRequest setAvailableSince(@Nullable Date availableSince) {
```
Comment (teryk) [inline sonar-ws/src/main/java/org/sonarqube/ws/client/rule/SearchWsRequest.java]: are you sure you want Date class? It's not clear that clear for me

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java (hunk 2 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java]: Hum, good catch, it's a mistake

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java]: In fact, this method is used here and in RuleQueryFactory, so it should be better to move it to RulesWsParameters, WDYT ?

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleQueryFactory.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+@ServerSide
+public class RuleQueryFactory {
+
+  private final DbClient dbClient;
+
+  public RuleQueryFactory(DbClient dbClient) {
+    this.dbClient = dbClient;
+  }
+
+  public RuleQuery createQuery(Request request) {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleQueryFactory.java]: `create` or `createRuleQuery` or `createFromWsRequest`

Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleQueryFactory.java]: I would document the check on the quality profile

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleQueryFactory.java]: createRuleQuery sounds good to me

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java (hunk 2 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java]: I'm not sure what you're talking about. Which method? I'm surprised the SearchWsRequest is used in BulkXXXAction

**Comment by teryk** [comment]
minor feedbacks and LGTM

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleQueryFactory.java (hunk 4 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleQueryFactory.java]: Ok, I'll add some Javadoc

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java (hunk 2 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java]: Let's talk about this offline

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java]: As discussed, I will reactivate this commented line, and remove the copy of this method from RuleQueryFactory

**Pull request merged**